indico/modules/events/surveys/controllers/management.py

```diff
@@ -259,3 +261,50 @@ def _render_questionnaire(survey):
     tpl = get_template_module('events/surveys/_questionnaire.html')
     form = make_survey_form(survey.questions)()
     return tpl.render_questionnaire(survey, form)
+
+
+class RHExportSubmissions(RHManageSurveyBase):
+    """Export submissions from the survey to a CSV file"""
+
+    def _process(self):
+        if not self.survey.submissions:
+            flash(_('There are no submissions in this survey'))
+            return redirect(url_for('.manage_survey', self.survey))
+
+        csv_file = CSVSubmissionsExporter(self.survey).generate()
+        delete_file.apply_async(args=[csv_file.name], countdown=3600)
+        return send_file('submissions-{}.csv'.format(self.survey.id), csv_file.name, 'text/csv')
+
+
+class RHSurveySubmissionBase(RHManageSurveysBase):
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.submission
+        }
+    }
+
+    def _checkParams(self, params):
+        RHManageSurveysBase._checkParams(self, params)
+        self.submission = SurveySubmission.get(request.view_args['submission_id'])
+
+
+class RHDeleteSubmission(RHSurveySubmissionBase):
+    """Remove a given submission"""
+
+    def _process(self):
+        db.session.delete(self.submission)
+        flash(_('Submission has been successfully removed'), 'success')
```

Comment:
if you delete the row instead of reloading the page we probably don't need this flash message anymore